backend/src/main/java/com/redhat/cloud/notifications/recipients/rbac/RbacRecipientUsersProvider.java

```diff
@@ -0,0 +1,81 @@
+package com.redhat.cloud.notifications.recipients.rbac;
+
+import com.redhat.cloud.notifications.recipients.RecipientUsersProvider;
+import com.redhat.cloud.notifications.recipients.User;
+import com.redhat.cloud.notifications.routers.models.Page;
+import io.smallrye.mutiny.Multi;
+import io.smallrye.mutiny.Uni;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.eclipse.microprofile.rest.client.inject.RestClient;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+@ApplicationScoped
+public class RbacRecipientUsersProvider implements RecipientUsersProvider {
+
+    @Inject
+    @RestClient
+    RbacServiceToService rbacServiceToService;
+
+    @ConfigProperty(name = "recipient_provider.rbac.elements_per_page")
+    Integer rbacElementsPerPage;
+
+    @Override
+    public Uni<List<User>> getUsers(String accountId, boolean adminsOnly) {
+        return getWithPagination(
+                page -> rbacServiceToService
+                        .getUsers(accountId, adminsOnly, page * rbacElementsPerPage, rbacElementsPerPage)
+                        .subscribeAsCompletionStage()
+        );
+    }
+
+    public Uni<List<User>> getGroupUsers(String accountId, boolean adminOnly, UUID groupId) {
+        return rbacServiceToService.getGroup(accountId, groupId)
+                .onItem().transformToUni(rbacGroup -> {
+                    if (rbacGroup.isPlatformDefault()) {
+                        return getUsers(accountId, adminOnly);
+                    } else {
+                        return getWithPagination(
+                                page -> rbacServiceToService
+                                        .getGroupUsers(accountId, groupId, page * rbacElementsPerPage, rbacElementsPerPage)
+                                        .subscribeAsCompletionStage()
+                        )
+                        // getGroupUsers doesn't have an adminOnly param.
+                        .onItem().transform(users -> {
+                            if (adminOnly) {
+                                return users.stream().filter(User::isAdmin).collect(Collectors.toList());
+                            }
+
+                            return users;
+                        });
+                    }
+                });
+    }
+
+    private Uni<List<User>> getWithPagination(Function<Integer, CompletionStage<Page<RbacUser>>> fetcher) {
+        return Multi.createBy().repeating()
+                .completionStage(
+                        AtomicInteger::new,
+                        state -> fetcher.apply(state.getAndIncrement())
+                )
+                .until(page -> page.getData().isEmpty())
```

Comment:
This would concatenate all `Multi` retrieved after we're done repeating:
```suggestion
                .until(page -> page.getData().isEmpty())
                .onItem().disjoint()
```
Then the following `transform` would only process one `RbacUser` at a time and at the end of the stream you could simply `.collect().asList()`.

Reply:
I'm a bit confused here, I though we wanted to avoid using Multi whenever possible when we could use stream operations.
What I understood here (and below) is that I'm getting multiple lists (as I'm not using the `disjoint`) and in the end I'm building a big list with all of them. 

I suppose the readability is improved when using `disjoint`. 

So is better to use the `disjoint` here and process as a multi?

Reply:
Ignore my comment, I misread one line of the stream, sorry about that 😄

Reply:
My comment was based on the asumption that we were transforming a `Multi<Multi<RbacUser>>` but I was wrong so the comment is irrelevant.

We should avoid creating `Multi` when `Uni` or simple Java stream are enough to do the job indeed.